charms/istio-pilot/src/charm.py

```diff
@@ -634,6 +635,22 @@ def _gateway_namespace(self):
         """Returns the namespace of the Gateway we will create, which is the same as the model."""
         return self.model.name
 
+    @property
+    def _is_gateway_object_up(self):
+        """Return whether the gateway object exists."""
+        try:
+            self._lightkube_client.get(
+                GATEWAY_LIGHTKUBE_RESOURCE,
+                name=self._gateway_name,
+                namespace=self._gateway_namespace
+            )
+            return True
+        except ApiError as e:
+            # If the object is missing, return False.  For other errors, raise.
+            if e.status.code == 404:
```

Comment:
I have no idea tbh.  We've used this sort of check elsewhere and not seen a 403 come up, but that doesn't mean it can't.  I'm not sure how to handle it here though - do you have any ideas?